beginner_source/ddp_fault_tolerance.rst

```diff
@@ -0,0 +1,212 @@
+`Introduction <ddp_series_intro.html>`__ \|\| `What is DDP <ddp_theory.html>`__ \|\| `Single-node
+Multi-GPU training <ddp_multigpu.html>`__ \|\| **Fault
+Tolerance** \|\| `Multi-node
+training <../intermediate/ddp_multinode.html>`__ \|\| `mingpt training <../intermediate/ddp_minGPT.html>`__
+
+
+Fault-tolerant Distributed Training with ``torchrun``
+=====================================================
+
+Authors: `Suraj Subramanian <https://github.com/suraj813>`__
+
+In distributed training, a single process failure can
+disrupt the entire training job. Since the susceptibility for failure can be higher here, making your training
+script robust is particularly important here. You might also prefer your training job to be *elastic* i.e. the ability to increase or decrease the number
+of processes in realtime while running the training job.
```

Comment:
do we actually provide this realtime capability? i.e. if we set max nodes to 3 and min nodes to 1, and we start training with 1 node, and 1 more node comes alive, do we proactively pause training, re-rendezvous, and continue, or does there need to be some sort of failure that stops the training? 

cc @d4l3k

Reply:
From reading the docs, my sense is that all processes are terminated and restarted on all available nodes with re-rendezvous, to continue from the last available checkpoint. Is "realtime" inaccurate to describe this?

Reply:
rephrased to "ability to increase or decrease the number of processes during training."